service/src/test/java/com/ge/predix/controller/test/PolicyEvalWithGraphDbControllerIT.java

```diff
@@ -0,0 +1,300 @@
+import static com.ge.predix.acs.testutils.Constants.RESOURCE_SITE_1_ID;
+import static com.ge.predix.acs.testutils.Constants.RESOURCE_XFILE_0_ATTRS_0;
+import static com.ge.predix.acs.testutils.Constants.RESOURCE_XFILE_0_ID;
+import static com.ge.predix.acs.testutils.Constants.SUBJECT_GROUP_0_ATTRS_0;
+import static com.ge.predix.acs.testutils.Constants.SUBJECT_GROUP_0_ID;
+import static com.ge.predix.acs.testutils.Constants.SUBJECT_GROUP_1_ATTRS_0;
+import static com.ge.predix.acs.testutils.Constants.SUBJECT_GROUP_1_ID;
+import static com.ge.predix.acs.testutils.Constants.SUBJECT_GROUP_2_ATTRS_0;
+import static com.ge.predix.acs.testutils.Constants.SUBJECT_GROUP_2_ID;
+import static com.ge.predix.acs.testutils.Constants.SUBJECT_USER_1_ATTRS_0;
+import static com.ge.predix.acs.testutils.Constants.SUBJECT_USER_1_ID;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.net.URLEncoder;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.web.context.WebApplicationContext;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.ge.predix.acs.model.Attribute;
+import com.ge.predix.acs.model.Effect;
+import com.ge.predix.acs.model.PolicySet;
+import com.ge.predix.acs.privilege.management.PrivilegeManagementService;
+import com.ge.predix.acs.rest.BaseResource;
+import com.ge.predix.acs.rest.BaseSubject;
+import com.ge.predix.acs.rest.Parent;
+import com.ge.predix.acs.rest.PolicyEvaluationRequestV1;
+import com.ge.predix.acs.rest.PolicyEvaluationResult;
+import com.ge.predix.acs.rest.Zone;
+import com.ge.predix.acs.testutils.MockAcsRequestContext;
+import com.ge.predix.acs.testutils.MockMvcContext;
+import com.ge.predix.acs.testutils.MockSecurityContext;
+import com.ge.predix.acs.testutils.TestUtils;
+import com.ge.predix.acs.utils.JsonUtils;
+import com.ge.predix.acs.zone.management.ZoneService;
+
+@WebAppConfiguration
+@ContextConfiguration("classpath:controller-tests-context.xml")
+@ActiveProfiles(profiles = { "h2", "public", "simple-cache", "titan" })
+public class PolicyEvalWithGraphDbControllerIT extends AbstractTestNGSpringContextTests {
```

Comment:
- Consider re-using the subject/resource graphs in GraphSubjectRepositoryTest & GraphResourceRepositoryTest, for ease of understanding. Same code can be used, and any nodes added in this test, if needed.